kbhff/test/login_test.py

```diff
@@ -90,3 +90,15 @@ def test_unverifiedUserWithPasswordFirstLogin(driver, gmail, unverified_user_via
     assert_username_prefilled(user["email"], driver)
     assert_text_on_page("Din konto er nu aktiveret og du kan logge ind", driver)
 
+#Test specification number 5
+def test_unverifiedUserWithPasswordFirstLoginIncorrectCode(driver, unverified_user_via_webform):
+    user = unverified_user_via_webform
+
+    navigate_to_page("login", driver)
+    try_login(driver, user["email"], user["password"])
+    assert_current_page_is("bekraeft_konto", driver)
+    wrong_token = "111aaaaa"
+    input_verification_token(wrong_token, driver)
+    assert_current_page_is("login", driver)
+    assert_username_not_prefilled("", driver)

```

Comment:
I'm confused why this works; based on the code in `assert_username_not_prefilled` it looks like this is testing that the username field is _not_ empty. (In particular, I think it would pass even if the field _were_ pre-filled with the username.)  Is this because there's a greyed-out field hint (e.g. "Username")?

Reply:
Yes, This started out as `assert_username_is_prefilled("", driver)` and when I noticed that the filed is indeed not empty even if not prefilled, I changed the name of the function but forgot to change the arguments. As per above, I think `assert_username_not_prefilled(driver)` is the way to go even if more brittle.